### PR TITLE
Generate the Semantic Colours as class members in Swift.

### PR DESCRIPTION
**Pull request by pixlwave**
The PR makes 2 changes to the Swift colour token generation (and can be reviewed commit-by-commit):
- Strips the `color` prefix from the colour tokens given they're all namespaced anyway.
- Splits up the generated Core/Semantic tokens and uses class members for the Semantic tokens instead of them being static.

### CODE REVIEW

File: src/configs/getIOSConfig.ts [hunk 1]
```diff
@@ -50,15 +56,31 @@ export function getIOSColorConfig(theme: Theme): Platform {
...
+        },
+        destination: "CompoundColorTokens.swift",
+        format: "swift/class-members",
+        options: {
+          showFileHeader: false,
+          outputReferences: true,
+          import: ["SwiftUI"],
+          objectType: "class",
+          accessControl: "public",
+          referenceClass: "CompoundCoreColorTokens"
```
Comment (pixlwave) [inline src/configs/getIOSConfig.ts:82]: This is somewhat of a hack…

File: src/formats/templates/swift/class-members.template [hunk 2]
```diff
@@ -21,13 +21,18 @@
...
-    return 'import ' + item
+    return `import ${item}`
   }).join('\n')
 %>
 
 <%= options.accessControl %> <%= options.objectType %> <%= file.className %> {
     <%= allProperties.map(function(prop) {
-      return options.accessControl + ' let ' + formatProperty(prop);
+      let formatted = formatProperty(prop);
+      if (options.referenceClass) {
```
Comment (pixlwave) [inline src/formats/templates/swift/class-members.template:31]: … but it works well enough to access the references from the correct class.

File: assets/ios/swift/CompoundCoreColorTokens.swift [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    public static let gray700 = Color("gray700", bundle: Bundle.module) 
+    public static let gray600 = Color("gray600", bundle: Bundle.module) 
+    public static let gray500 = Color("gray500", bundle: Bundle.module) 
+    public static let gray400 = Color("gray400", bundle: Bundle.module) 
+    public static let gray300 = Color("gray300", bundle: Bundle.module) 
+    public static let gray200 = Color("gray200", bundle: Bundle.module) 
+    public static let gray100 = Color("gray100", bundle: Bundle.module) 
+    public static let themeBg = Color("themeBg", bundle: Bundle.module) 
+    public static let bgSubtleSecondaryLevel0 = themeBg
+    public static let bgCanvasDefaultLevel1 = gray300
```
Comment (pixlwave) [inline assets/ios/swift/CompoundCoreColorTokens.swift:319]: These 2 aren't used in the app as they're asymmetric and we still need to figure out how to generate them correctly.

File: src/transforms/swift/tokenTICamel.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ import { Transform } from "style-dictionary/types/Transform";
 export default {
   type: 'name',
   matcher: function (token) {
-    return token.type === "icon";
+    return token.type == "color" || token.type === "icon";
```
Comment (robintown) [inline src/transforms/swift/tokenTICamel.ts:27]:
```suggestion
    return token.type === "color" || token.type === "icon";
```

File: src/configs/getIOSConfig.ts [hunk 5]
```diff
@@ -51,15 +56,31 @@ export function getIOSColorConfig(theme: Theme): Platform {
...
         options: {
           showFileHeader: false,
           outputReferences: true,
           import: "SwiftUI",
         },
+        className: "CompoundCoreColorTokens",
+      },
+      {
+        filter: function(token: TransformedToken) {
+          return token.type == 'color' && isNotCoreColor.matcher(token);
```
Comment (robintown) [inline src/configs/getIOSConfig.ts:72]:
```suggestion
          return token.type === 'color' && isNotCoreColor.matcher(token);
```

**Review by robintown — APPROVED** [review]
Looks good, and sorry that these little things aren't caught automatically by a linter. Hopefully I can get linting set up today!

**Pull request merged**